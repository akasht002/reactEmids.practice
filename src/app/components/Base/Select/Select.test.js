import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import {SelectBox} from './index';

Enzyme.configure({ adapter: new Adapter() })

const SelectPlan = [
  { label: "Plan One", value: 1, },
  { label: "Plan Two", value: 2, },
  { label: "Plan Three", value: 3, },
  { label: "Plan Four", value: 4, },
  { label: "Plan Five", value: 5, },
  { label: "Plan Six", value: 6, },
  { label: "Plan Seven", value: 7, },
  { label: "Plan Eight", value: 8, },
  { label: "Plan Nine", value: 9, },
];

describe('SelectBox', () => {
  it('should return correct component', () => {
    const wrapper = shallow(
      <SelectBox options={SelectPlan} id='1' placeholder='Select' value='John Doe'></SelectBox>,
    )
    expect(wrapper).toBeDefined();
  })
});



