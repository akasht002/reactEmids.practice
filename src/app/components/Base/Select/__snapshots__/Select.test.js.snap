// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox Component Jest 1`] = `
<div
  className="Select is-clearable is-searchable Select--single"
>
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="Select-multi-value-wrapper"
      id="react-select-2--value"
    >
      <div
        className="Select-placeholder"
      >
        Select
      </div>
      <div
        className="Select-input"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <input
          aria-activedescendant="react-select-2--value"
          aria-expanded="false"
          aria-haspopup="false"
          aria-owns=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          role="combobox"
          style={
            Object {
              "boxSizing": "content-box",
              "width": "5px",
            }
          }
          value=""
        />
        <div
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "scroll",
              "position": "absolute",
              "top": 0,
              "visibility": "hidden",
              "whiteSpace": "pre",
            }
          }
        >
          
        </div>
      </div>
    </div>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="Select-arrow"
        onMouseDown={[Function]}
      />
    </span>
  </div>
</div>
`;
